packages/web/components/earn/filters/top-filters.tsx

```diff
@@ -123,14 +123,18 @@ export const TopFilters = ({
   return (
     <div className="flex flex-col gap-5 px-10 py-8 1.5xs:px-7 1.5xs:py-7">
       <div className="flex flex-wrap items-center justify-between gap-7 2xl:gap-10 1.5xl:gap-4 lg:hidden">
-        <RadioWithOptions
-          disabled={tokenHolderSwitchDisabled}
-          mode="primary"
-          variant="large"
-          value={tokenHolder}
-          onChange={(value) => setFilter("tokenHolder", value)}
-          options={tokenFilterOptions}
-        />
+        {!tokenHolderSwitchDisabled ? (
+          <RadioWithOptions
+            disabled={tokenHolderSwitchDisabled}
+            mode="primary"
+            variant="large"
+            value={tokenHolder}
+            onChange={(value) => setFilter("tokenHolder", value)}
+            options={tokenFilterOptions}
+          />
+        ) : (
+          false
```

Comment:
Why not use `!tokenHolderSwitchDisabled && <Component />`?

Reply:
`&&` operator can lead unexpected behaviours:
https://www.crocoder.dev/blog/react-conditional-rendering/

Reply:
interesting.